### PR TITLE
feat: added horizontal scrollbars for `UITable` elements.

### PR DESCRIPTION
**Pull request by s4my**

### CODE REVIEW

**Comment by nakst** [comment]
I fixed some edge cases in your commit for `UICode` here: [e870272](https://github.com/nakst/gf/commit/e8702723f158a50273f3b062900455e28af5bf22).
Can you update this commit to also handle the edge cases? Thanks

**Comment by s4my** [comment]
> I fixed some edge cases in your commit for `UICode` here: [e870272](https://github.com/nakst/gf/commit/e8702723f158a50273f3b062900455e28af5bf22). Can you update this commit to also handle the edge cases? Thanks

What were those edge cases ? your commit message is ambiguous. I also left a comment in the commit asking for a clarification.

**Comment by s4my** [comment]
Refactored the code, tell if it's all right with you.

File: luigi2.h [hunk 1]
```diff
@@ -3051,7 +3051,7 @@ int _UITableMessage(UIElement *element, UIMessage message, int di, void *dp) {
 			UIDrawControl(painter, row, UI_DRAW_CONTROL_TABLE_ROW | rowFlags, NULL, 0, 0, element->window->scale);
 
 			UIRectangle cell = row;
-			cell.l += UI_SIZE_TABLE_COLUMN_GAP * table->e.window->scale;
+			cell.l += UI_SIZE_TABLE_COLUMN_GAP - (table->hScroll ? (int64_t) table->hScroll->position : 0) * table->e.window->scale;
```
Comment (nakst) [inline luigi2.h:3054]: `UI_SIZE_TABLE_COLUMN_GAP` should be multiplied by the scale, not the scroll position.

File: luigi2.h [hunk 2]
```diff
@@ -3094,12 +3097,44 @@ int _UITableMessage(UIElement *element, UIMessage message, int di, void *dp) {
 				}
 			}
 		}
+		UIElementRefresh(element);
```
Comment (nakst) [inline luigi2.h:3100]: What is this for? Doesn't this cause the table to redraw itself every event?

**Comment by nakst** [comment]
Sorry it takes me so long to review pull requests, I am very busy at the moment.

File: luigi2.h (hunk 2 above)
Comment (s4my) [reply luigi2.h:3100]: I guess this is no longer needed, i fixed this problem this line solved before i'll just remove it.

File: luigi2.h (hunk 1 above)
Comment (s4my) [reply luigi2.h:3054]: this is needed to scroll the different entries horizontally, without this only the header is scrolled.

**Comment by s4my** [comment]
> Sorry it takes me so long to review pull requests, I am very busy at the moment.

It's all right, just you apologizing is more than enough.

**Pull request merged**